modules/cli/src/main/scala/scala/cli/ScalaCli.scala

```diff
@@ -23,7 +23,9 @@ object ScalaCli {
   // TODO: Remove this part once fix is released in os-lib (Issue #2585)
   if (scala.util.Try(os.Path(System.getProperty("user.home"))).isFailure) {
     System.err.println("Warning: user.home property is not set, setting it to user.dir")
-    System.setProperty("user.home", System.getProperty("user.dir"))
+    val homeDirOverride = Option(System.getenv("SCALA_CLI_HOME_DIR_OVERRIDE"))
```

Comment:
I suppose if we have an override, it should work arbitrarily.
as in, even if the `user.home` property is present, the env override should still be applied outside of this if clause.